exoscale/sks_agent_runner_node_csr_validation.go

```diff
@@ -104,30 +106,42 @@ func (r *sksAgentRunnerNodeCSRValidation) run(ctx context.Context) {
 					errorf("sks-agent: expected 1 certificate Subject Alternate Name DNS Name value, got %d", l)
 					continue
 				}
-				if l := len(parsedCSR.IPAddresses); l != 1 {
-					errorf("sks-agent: expected 1 certificate Subject Alternate Name IP Address value, got %d", l)
-					continue
-				}
 
 				instances, err := r.p.client.ListInstances(ctx, r.p.zone)
 				if err != nil {
 					errorf("sks-agent: failed to list Compute instances: %v", err)
 					continue
 				}
 
+				csrOK := false
 				for _, instance := range instances {
 					if *instance.Name == parsedCSR.DNSNames[0] {
-						if instance.PublicIPAddress.Equal(parsedCSR.IPAddresses[0]) {
-							ok = true
+						var addrs, addrsCSR []string
+
+						addrs = append(addrsCSR, instance.PublicIPAddress.String())
+
+						if instance.IPv6Enabled != nil && *instance.IPv6Enabled {
+							addrs = append(addrs, instance.IPv6Address.String())
+						}
+
+						for _, ip := range parsedCSR.IPAddresses {
+							addrsCSR = append(addrsCSR, ip.String())
+						}
+
+						sort.Strings(addrs)
+						sort.Strings(addrsCSR)
+
+						if !reflect.DeepEqual(addrs, addrsCSR) {
+							errorf("sks-agent: CSR %s Node IP address doesn't match corresponding "+
+								"Compute instance public interface IP address, %q, got %q", csr.Name, addrs, parsedCSR.IPAddresses)
+
 							break
 						}
```

Comment:
The CSR object contains ip addresses from the kubelet configuration (e.g. Kubelet `--node-ip` argument).

When not specified it defaults to eth0 ipv4 IIRC.
But other valid use cases can also be:
- only the public IPv6
- only a single private IP from a privnet
- all ip addresses assigned to the instance (what this PR handles).

In order to handle all use cases, `addrsCSR` must be a subset of `addrs`, not necessarily equal